src/QtLocationPlugin/QMLControl/OfflineMap.qml

```diff
@@ -368,6 +369,23 @@ Item {
                         visible:        _esriFact ? _esriFact.visible : false
                         font.pointSize: _adjustableFontPointSize
                     }
+
+                    Item { width: 1; height: 1; visible: _customURLFact ? _customURLFact.visible : false }
+                    QGCLabel { text: qsTr("Custom Map URL"); visible: _customURLFact ? _customURLFact.visible : false }
+                    FactTextField {
+                        fact:               _customURLFact
+                        visible:            _customURLFact ? _customURLFact.visible : false
```

Comment:
You can have a property for visibility to avoid the same code in multiple places

Reply:
Do you have an example of this somewhere in code?
I am interested in using it, I just have not had time to familiarise myself with QML, so I have been borrowing from existing code.

Reply:
The if check for fact being null shouldn't be needed here as far as I can tell. This code is pretty old, not sure why is has all the null checks? Most of this stuff will never be null.